controllers/snapshot/snapshot_adapter.go

```diff
@@ -306,6 +306,17 @@ func (a *Adapter) EnsureCreationOfEphemeralEnvironments() (controller.OperationR
 					"integrationTestScenario.Name", integrationTestScenario.Name)
 				return controller.StopProcessing()
 			}
+			if clienterrors.IsInvalid(err) {
+				a.logger.Error(err, "environment or deploymentTargetClass referenced by integrationScenario is invalid", "integrationTestScenario.Name", integrationTestScenario.Name)
```

Comment:
We can provide what exactly is wrong, which kind and name

```diff
diff --git a/controllers/snapshot/snapshot_adapter.go b/controllers/snapshot/snapshot_adapter.go
index 86cc817..9518bb3 100644
--- a/controllers/snapshot/snapshot_adapter.go
+++ b/controllers/snapshot/snapshot_adapter.go
@@ -18,6 +18,7 @@ package snapshot
 
 import (
        "context"
+       "errors"
        "fmt"
        "reflect"
        "strings"
@@ -307,10 +308,20 @@ func (a *Adapter) EnsureCreationOfEphemeralEnvironments() (controller.OperationR
                                return controller.StopProcessing()
                        }
                        if clienterrors.IsInvalid(err) {
-                               a.logger.Error(err, "environment or deploymentTargetClass referenced by integrationScenario is invalid", "integrationTestScenario.Name", integrationTestScenario.Name)
+                               var (
+                                       errKind   string = "<unknown>"
+                                       errName   string = "<unknown>"
+                                       statusErr *clienterrors.StatusError
+                               )
+
+                               if ok := errors.As(err, &statusErr); ok {
+                                       errKind = statusErr.Status().Details.Kind
+                                       errName = statusErr.Status().Details.Name
+                               }
+                               a.logger.Error(err, fmt.Sprintf("Object %s (%s) referenced by integrationScenario is invalid", errName, errKind), "integrationTestScenario.Name", integrationTestScenario.Name)
                                if !gitops.IsSnapshotMarkedAsFailed(a.snapshot) {
                                        _, err = gitops.MarkSnapshotAsFailed(a.client, a.context, a.snapshot,
-                                               fmt.Sprintf("The environment or deploymentTargetClass associated with snapshot %s is invalid.", a.snapshot.Spec.Application))
+                                               fmt.Sprintf("Object %s (%s) associated with snapshot %s is invalid: %s.", errName, errKind, a.snapshot.Spec.Application, err))
                                        if err != nil {
                                                a.logger.Error(err, "Failed to Update Snapshot status")
                                        }

```

Reply:
s/Object/Resource

Reply:
This was useful, thank you. Should be fixed in [b674ec](https://github.com/redhat-appstudio/integration-service/pull/421/commits/b674ecd242f54c9700dd14f99c04946a91f6af72)